libraries/lib-effects/MixAndRender.cpp

```diff
@@ -150,23 +128,17 @@ TrackListHolder MixAndRender(const TrackIterRange<const WaveTrack> &trackRange,
          if (blockLen == 0)
             break;
 
-         if (mono) {
-            auto buffer = mixer.GetBuffer();
-            mixLeft->Append(buffer, format, blockLen, 1, effectiveFormat);
-         }
-         else {
-            auto buffer = mixer.GetBuffer(0);
-            mixLeft->Append(buffer, format, blockLen, 1, effectiveFormat);
-            buffer = mixer.GetBuffer(1);
-            mixRight->Append(buffer, format, blockLen, 1, effectiveFormat);
+         for(auto channel : mix->Channels())
+         {
+            auto buffer = mixer.GetBuffer(channel->GetChannelIndex());
```

Comment:
In the present transitional state of the source tree, this needs to be `ReallyGetChannelIndex`, I think